src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -293,9 +293,9 @@ class BlockImporter(
     }
   }
 
-  private def importWith(importFuture: => Future[NewBehavior])(state: ImporterState): Unit = {
+  private def importWith(importTask: => Task[NewBehavior])(state: ImporterState): Unit = {
```

Comment:
It would be nice to remove the "by name" parameter passing and make sure that everywhere `importWith` is called it's doing logging in a monadic way rather than as a side effect of creating the task.